src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1293 @@
+										if (increaseButton != null)
+										{
+											increaseButton.IsEnabled = false;
+										}
+									}
+								}
+							}
+						}
+						else
+						{
+							// Case: TabWidthMode "Compact" or "FitToContent"
+							tabColumn.MaxWidth = availableWidth;
+							tabColumn.Width = GridLengthHelper.FromValueAndType(1.0, GridUnitType.Auto);
+							var listview = m_listView;
+							if (listview != null)
+							{
+								listview.MaxWidth = availableWidth;
+
+								// Calculate if the scroll buttons should be visible.
+								var itemsPresenter = m_itemsPresenter;
+								if (itemsPresenter != null)
+								{
+									var visible = itemsPresenter.ActualWidth > availableWidth;
+									ScrollViewer.SetHorizontalScrollBarVisibility(listview, visible
+										? ScrollBarVisibility.Visible
+										: ScrollBarVisibility.Hidden);
+									if (visible)
+									{
+										UpdateScrollViewerDecreaseAndIncreaseButtonsViewState();
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+
+
+			if (shouldUpdateWidths || TabWidthMode != TabViewWidthMode.Equal)
+			{
+				foreach (var item in TabItems)
+				{
+					// Set the calculated width on each tab.
+					var tvi = item as TabViewItem;
+					if (tvi == null)
+					{
+						tvi = ContainerFromItem(item) as TabViewItem;
+					}
+
+					if (tvi != null)
+					{
+						tvi.Width = tabWidth;
+					}
+				}
+			}
+		}
+
+		void UpdateSelectedItem()
+		{
+			var listView = m_listView;
+			if (listView != null)
+			{
+				var tvi = SelectedItem as TabViewItem;
+				if (tvi == null)
+				{
+					tvi = ContainerFromItem(SelectedItem) as TabViewItem;
+				}
+
+				if (tvi != null)
+				{
+					listView.SelectedItem = tvi;
+
+					// Setting ListView.SelectedItem will not work here in all cases.
+					// The reason why that doesn't work but this does is unknown.
+					tvi.IsSelected = true;
+				}
+			}
+		}
+
+		private void UpdateSelectedIndex()
+		{
+			var listView = m_listView;
+			if (listView != null)
+			{
+				listView.SelectedIndex = SelectedIndex;
+			}
+		}
+
+		private DependencyObject ContainerFromItem(object item)
+		{
+			var listView = m_listView;
+			if (listView != null)
+			{
+				return listView.ContainerFromItem(item);
+			}
+			return null;
+		}
+
+		internal DependencyObject ContainerFromIndex(int index)
+		{
+			var listView = m_listView;
+			if (listView != null)
+			{
+				return listView.ContainerFromIndex(index);
+			}
+			return null;
+		}
+
+		private object ItemFromContainer(DependencyObject container)
+		{
+			var listView = m_listView;
+			if (listView != null)
+			{
+				return listView.ItemFromContainer(container);
+			}
+			return null;
+		}
+
+		private int GetItemCount()
+		{
+			var itemsSource = TabItemsSource;
+			if (itemsSource != null)
+			{
+				var iterable = itemsSource as IEnumerable<DependencyObject>; //TODO:MZ:Verify cast is correct
+				if (iterable != null)
+				{
+					//int i = 1;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)